### ISSUE TITLE
False positive: indentation of text block inside annotation array parameters

### ISSUE THREAD

**Issue by lrozenblyum** [checkstyle/checkstyle#18615]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```cmd
E:\temp\CheckstyleReport>type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation" />
  </module>
</module>

E:\temp\CheckstyleReport>type Test.java
public class Test {
    @interface Anno {
        String [] value();
    }

    @Anno({
        """
        example
        """
    })
    void method() {
    }
}
E:\temp\CheckstyleReport>set RUN_LOCALE="-Duser.language=en -Duser.country=US"

E:\temp\CheckstyleReport>java %RUN_LOCALE% -jar checkstyle-13.0.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] E:\temp\CheckstyleReport\Test.java:9:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

**Describe what you expect in detail.**
End of text block formatting shouldn't cause violation of indentation rules.

**Comment by syabru** [comment]
See also #17223

**Comment by M-SaaD-H** [comment]
I'm on it.

**Comment by Anushreebasics** [comment]
hey @M-SaaD-H I am already working on this

**Comment by M-SaaD-H** [comment]
Hi @Anushreebasics ,
You have to comment on the issue before starting to work. Please keep that in mind for your further issues.

Do you have a fix ready for this? If not I have one, and I can raise the PR.

**Comment by Anushreebasics** [comment]
I was working on it but okay sure, you can go ahead

**Comment by romani** [comment]
fixed by @M-SaaD-H .
